lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -462,9 +462,37 @@ class MongoClientInterface {
         });
     }
 
+    /**
+     * Returns the suitable mongo operation to perform
+     * depending on the version being put
+     * @param {Boolean} isDeleteMarker isDeleteMarker tag
+     * @param {String} vFormat vFormat of bucket
+     * @param {Object} filter filter to get master
+     * @param {Object} update value to update master with
+     * @param {Boolean} upsert if upserting is needed
+     * @return {Object} mongo operation
+     */
+    updateDeleteMaster(isDeleteMarker, vFormat, filter, update, upsert) {
```

Comment:
I don't think having a different file for each version is a good idea in this point in time, because that will introduce a lot of code duplication.
The logic between v0 and v1 is actually pretty similar, all the changes introduced in this PR are very slight, for the most part we only add an extra step where we check if the master needs to be deleted.